activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -575,7 +575,7 @@ def visit_Arel_Table(o, collector)
           end
 
           if o.table_alias
-            collector << " " << quote_table_name(o.table_alias)
+            collector << " AS " << quote_table_name(o.table_alias)
```

Comment:
I'm struggling to see how this fixes the issue described at https://github.com/rails/rails/issues/48775.

> ### Expected behavior
> The table alias is valid on SELECTs, but not on INSERT, UPDATE, or DELETE statements.
> The example is with sqlite, but [MySQL has similar rules where it specifically asks for a table name](https://dev.mysql.com/doc/refman/8.0/en/insert.html).
> 
> I would expect Arel to NOT use the alias on insert, update, and delete statements

Reply:
This fixes your issue by making the INSERT, UPDATE, or DELETE statements function when using an alias.  As I was looking into this, this seemed a more succinct and viable fix to your problem than overriding the various statements in the `Arel::Visitors::Visitor` `ToSql` class just to simply eliminate using an alias at all in the SQL statements.  If you run your code from the example with the change you will find it now works.

Reply:
Looks like it does fix SQLite and fixes the update and delete for MySQL, but it fails on the insert:
```rb
  def test_insert_alias
    Post.arel_table.table_alias = "p"
    e = assert_raises(ActiveRecord::StatementInvalid) do
      Post.create!
    end
    assert_equal('INSERT INTO `posts` AS `p` VALUES ()', e.sql)
    assert_equal("Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'AS `p` VALUES ()' at line 1", e.message)
  end
```

Reply:
That's fair, I can add an integration test.  Looking into the MYSQL INSERT issue as well.

Reply:
Thanks for pointing out the issue with INSERT in MySQL.  I now have a version that addresses that and works with all 3 Methods and all 3 DB types(SQLite, PostgreSQL, and MySQL).  I'll add integration tests later this evening when I have more time.

Reply:
Wonderful. Thank you for your hard work